x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java

```diff
@@ -251,17 +266,42 @@ public void flushJob(JobTask jobTask, FlushJobParams params, ActionListener<Flus
         });
     }
 
+    /**
+     * Request temporary storage to be used for the job
+     *
+     * @param jobTask
+     *            The job task
+     * @param requestedSizeAsBytes
+     *            requested size
+     * @return a Path to local storage or null if storage is not available
+     */
+    public Path tryGetTmpStorage(JobTask jobTask, long requestedSizeAsBytes) {
```

Comment:
I think we could pass the requested size as a `ByteSizeValue` and only convert at the lower level. This will make the API less error prone.